main/src/ca/uwaterloo/flix/language/phase/OccurrenceAnalyzer.scala

```diff
@@ -19,33 +19,44 @@ package ca.uwaterloo.flix.language.phase
 import ca.uwaterloo.flix.api.Flix
 import ca.uwaterloo.flix.language.CompilationMessage
 import ca.uwaterloo.flix.language.ast.LiftedAst.Expression
-import ca.uwaterloo.flix.language.ast.OccurrenceAst.{Occur, DefContext}
 import ca.uwaterloo.flix.language.ast.OccurrenceAst.Occur._
-import ca.uwaterloo.flix.language.ast.Symbol.{LabelSym, VarSym}
+import ca.uwaterloo.flix.language.ast.OccurrenceAst.{DefContext, Occur}
+import ca.uwaterloo.flix.language.ast.Symbol.{DefnSym, LabelSym, VarSym}
 import ca.uwaterloo.flix.language.ast.{LiftedAst, OccurrenceAst, Symbol}
 import ca.uwaterloo.flix.util.Validation.ToSuccess
 import ca.uwaterloo.flix.util.{ParOps, Validation}
 
 /**
- * The occurrence analyzer collects information on variable usage
- * Marks a variable as Dead if it is not used, Once if it is used exactly once and Many otherwise
+ * The occurrence analyzer collects information on variable and function usage and calculates the weight of the expressions
+ * Marks a variable or function as Dead if it is not used, Once if it is used exactly once and Many otherwise
  */
 object OccurrenceAnalyzer {
 
+  object OccurInfo {
+    val EmptyOccur: OccurInfo = OccurInfo(Map.empty, Map.empty, 0)
+
+    val BaseOccur: OccurInfo = OccurInfo(Map.empty, Map.empty, 1)
+  }
+
+  case class OccurInfo(varOccurrence: Map[VarSym, OccurrenceAst.Occur], defOccurrence: Map[DefnSym, OccurrenceAst.Occur], codeSize: Int) {
```

Comment:
Add documentation.

Usually these classes are sorted, i.e. defOccurrence would be before varOccurrence. You could also consider calling these just `defs` and `vars`.